File: src/examples/createSignatureOrder/signatureAppearance.py
from criipto_signatures import CriiptoSignaturesSDKSync
from criipto_signatures.models import (
  CreateSignatureOrderInput,
  DocumentInput,
  PadesDocumentInput,
  DocumentStorageMode,
  SignatureAppearanceInput,
  EvidenceProviderInput,
  CriiptoVerifyProviderInput,
)

client = CriiptoSignaturesSDKSync(
  "{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}"
)

# Create signature order
signatureOrder = client.createSignatureOrder(
  CreateSignatureOrderInput(
    title="Python sample",
    documents=[
      DocumentInput(
        pdf=PadesDocumentInput(
          title="My document",
          blob=bytes("...", "utf-8"),  # bytes object, or a base64 encoded string
          storageMode=DocumentStorageMode.Temporary,
        ),
      )
    ],
    evidenceProviders=[
      EvidenceProviderInput(
        criiptoVerify=CriiptoVerifyProviderInput(scope="openid ssn")
      )
    ],
    signatureAppearance=SignatureAppearanceInput(
      identifierFromEvidence=[
        "cprNumberIdentifier",
        "dk:gov:saml:attribute:CprNumberIdentifier",
        "ssn",
        "http://schemas.grean.id/claims/se/ssn",
        "socialno",
        "http://schemas.grean.id/claims/no/socialno",
      ]
    ),
  )
)

print(signatureOrder.id)

File: src/examples/createSignatureOrder/signatureAppearance.cs
using Criipto.Signatures;
using Criipto.Signatures.Models;

using (var client = new CriiptoSignaturesClient("{YOUR_CRIIPTO_CLIENT_ID}", "{YOUR_CRIIPTO_CLIENT_SECRET}"))
{
    // Setup document input
    var documents = new List<DocumentInput>{
        new DocumentInput {
            pdf = new PadesDocumentInput {
                title = "Dotnet Sample",
                blob = new Byte[64], // Should be the bytes of a PDF file
                storageMode = DocumentStorageMode.Temporary
            }
        }
    };


    var evidenceProviders = new List<EvidenceProviderInput>() {
      new EvidenceProviderInput {
        criiptoVerify = new CriiptoVerifyProviderInput() {
          scope = "openid ssn",

        }
      }
    };

    // Setup signature order input
    var createSignatureOrderInput = new CreateSignatureOrderInput
    {
        title = "Dotnet Sample",
        documents = documents,
        evidenceProviders = evidenceProviders,
        signatureAppearance = new SignatureAppearanceInput()
        {
            identifierFromEvidence = new List<string>() {
                "cprNumberIdentifier",
                "dk:gov:saml:attribute:CprNumberIdentifier",
                "ssn",
                "http://schemas.grean.id/claims/se/ssn",
                "socialno",
                "http://schemas.grean.id/claims/no/socialno"
            }
        }
    };

    // Create signature order
    var signatureOrder = await client.CreateSignatureOrder(createSignatureOrderInput);
}